### PR TITLE
[CN-812] Rename secret to secretName for ManagementCenter and HotBackup

### PR DESCRIPTION
**Pull request by SeriyBg**
## User Impact

The field licenseKeySecret in ManagementCenterSpec was renamed to licenseKeySecretName.
The field secret in HotBackupSpec was renamed to secretName.

### CODE REVIEW

File: test/e2e/config/managementcenter/config.go [hunk 1]
```diff
@@ -125,9 +125,9 @@ var (
 				Labels:    lbls,
 			},
 			Spec: hazelcastv1alpha1.ManagementCenterSpec{
-				Repository:       *mcRepo,
-				Version:          "not-exists",
-				LicenseKeySecret: licenseKey(ee),
+				Repository:                 *mcRepo,
+				Version:                    "not-exists",
+				DeprecatedLicenseKeySecret: licenseKey(ee),
```
Comment (semihbkgr) [inline test/e2e/config/managementcenter/config.go:130]:
```suggestion
				LicenseKeySecretName: licenseKey(ee),
```

File: test/e2e/helpers_test.go [hunk 2]
```diff
@@ -905,7 +905,7 @@ func restoreConfig(hotBackup *hazelcastcomv1alpha1.HotBackup, useBucketConfig bo
 		return hazelcastcomv1alpha1.RestoreConfiguration{
 			BucketConfiguration: &hazelcastcomv1alpha1.BucketConfiguration{
 				BucketURI:  hotBackup.Status.GetBucketURI(),
-				SecretName: hotBackup.Spec.Secret,
+				SecretName: hotBackup.Spec.SecretName,
```
Comment (semihbkgr) [inline test/e2e/helpers_test.go:908]:
```suggestion
				SecretName: hotBackup.Spec.GetSecretName(),
```

**Review by semihbkgr — COMMENTED** [review]
Maybe we can consider to set values to new fields taking values from their corresponding deprecated fields in reconciler loop.

**Comment by SeriyBg** [comment]
> Maybe we can consider to set values to new fields taking values from their corresponding deprecated fields in reconciler loop.

This is a good option. But for this, I think we also need to have the `defaulter` webhook. So I will create a JIRA ticket to implement it in a separate task.

**Pull request merged**